packages/hardhat-core/test/internal/hardhat-network/provider/modules/web3.ts

```diff
@@ -28,7 +29,9 @@ describe("Web3 module", function () {
       describe("web3_sha3", async function () {
         it("Should return the keccak256 of the input", async function () {
           const data = "0x123a1b238123";
-          const hashed = bufferToRpcData(keccak256(toBuffer(data)));
+          const hashed = bufferToRpcData(
+            arrToBufArr(keccak256(bufArrToArr(toBuffer(data))))
```

Comment:
I'm pretty sure this is not needed. `Buffer`s are `Uint8Array`s.

Reply:
Yes but `Uint8Array`s are not `Buffer`s :sweat_smile: And since `Uint8Array.prototype.toString` doesn't accept an encoding, you have to convert it to Buffer.

We can create some helper for this, but it would be just for readability, the number of conversions will be the same. We could write our own `uint8ArrayToHex` but we should benchmark it to verify that it's indeed faster (and my recent attempt to write a faster `bufferToBigInt` makes me doubt that it will be).

Reply:
Oh, I was getting it wrong. Thanks